Libraries/ReactNative/renderApplication.js

```diff
@@ -13,29 +13,37 @@
 const AppContainer = require('AppContainer');
 const React = require('React');
 const ReactFabricIndicator = require('ReactFabricIndicator');
+const RootViewLayout = require('RootViewLayout');
 
 const invariant = require('invariant');
 
+import type {LayoutContext} from 'RootViewLayout';
+
 // require BackHandler so it sets the default handler that exits the app if no listeners respond
 require('BackHandler');
 
 function renderApplication<Props: Object>(
   RootComponent: React.ComponentType<Props>,
   initialProps: Props,
   rootTag: any,
+  initialLayoutContext: LayoutContext,
   WrapperComponent?: ?React.ComponentType<*>,
   fabric?: boolean,
   showFabricIndicator?: boolean,
 ) {
   invariant(rootTag, 'Expect to have a valid rootTag, instead got ', rootTag);
 
   let renderable = (
-    <AppContainer rootTag={rootTag} WrapperComponent={WrapperComponent}>
-      <RootComponent {...initialProps} rootTag={rootTag} />
-      {fabric === true && showFabricIndicator === true ? (
-        <ReactFabricIndicator />
-      ) : null}
-    </AppContainer>
+    <RootViewLayout.Manager
```

Comment:
I wonder if we should just bake this new root layout functionality into AppContainer since it already has subscriptions and context and state - that would reduce the "wrapper hell" we have going on. Ideally we could factor out the functionality into separate hook modules `useRootLayoutProvider` / `useRootLayout`?

Reply:
The reason for not having it inside `AppContainer` is that it is also used in modal here https://github.com/facebook/react-native/blob/007e00fa70d0e1d69f7148e901790d2092e15b8c/Libraries/Modal/Modal.js#L244

Reply:
Adding a `useRootLayout` hook sounds good, it think it will just be a simple wrapper around `useContext`.